portal-backend/depmap/compute/views.py

```diff
@@ -172,9 +179,16 @@ def post(self):
             # 2. which is dependent/independent, the matrix or the vector
             # 3. optionally, a list of cell line depmap ids
             assert slice_query is not None
-            query_series = data_access.get_slice_data(slice_query)
-            # remove missing entries because the intersection is assuming we have values for everything. perhaps this should be put into get_slice_data?
-            query_series = query_series[~query_series.isna()]
+            if slice_query_is_from_breadbox:
+                query_series = data_access.get_slice_data(slice_query)
+            else: 
+                # In this specific case, it's important to avoid the data_access interface because 
+                # breadbox legacy-dataset aliases don't work for lookups by entity_id 
+                # (which is still the slice format used by celfie/genomic associations)
+                query_series = interactive_utils.get_row_of_values_from_slice_id(
+                    query_id
+                )
```

Comment:
I had initially thought the source of the issue was that custom analysis was being called with a dataset that was aliased in breadbox and a legacy "entity_id" which doesn't consistently map onto the IDs breadbox uses for features (for some datasets they do match, for others they don't). My concern was that our aliasing was automatically trying to look up values in breadbox, when some of the old slice IDs only work with legacy datasets. 

In reality, I don't think anything is being called this way, so it's probably not an issue, but I updated this function just in case. Now, if anyone calls custom analysis with a non-breadbox slice ID, it will avoid the possibility of aliasing.